src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,321 @@
+        } else {
+            for (int i = 0; i < tasks.size(); i += 1) {
+                tasks.get(i).printTask(i);
+            }
+        }
+
+    }
+
+    public static TaskType identifyTask(String command) {
+        String commandType = command.split(" ")[0];
+
+        switch (commandType) {
+        case "list":
+            return TaskType.LIST;
+        case "mark":
+            return TaskType.MARK;
+        case "todo":
+        case "deadline":
+        case "event":
+            return TaskType.ADD;
+        case "bye":
+            return TaskType.BYE;
+        case "delete":
+            return TaskType.DELETE;
+        default:
+            return TaskType.ERROR;
+        }
+    }
+
+    private static void appendToFile(String taskType) throws IOException {
+        try {
+            FileWriter fw = new FileWriter(filePath, true);
+            int lastIndex = tasks.size() - 1;
+            String appendTask = tasks.get(lastIndex).getTask();
+
+            if (taskType.equals("todo")) {
+                fw.write("T | 0 | " + appendTask + System.lineSeparator());
+            } else if (taskType.equals("deadline")) {
+                fw.write("D | 0 | " + appendTask + System.lineSeparator());
+            } else {
+                fw.write("E | 0 | " + appendTask + System.lineSeparator());
+            }
+            fw.close();
+
+        } catch (IOException exception) {
+            System.out.println("Error writing to file.");
+        }
+    }
+
+    public static void load() throws FileNotFoundException {
+
+        File f = new File(filePath);
+
+        if (f.exists()) {
+            Scanner s = new Scanner(f);
+            while (s.hasNext()) {
+                String entry = s.nextLine();
+                String[] entries = entry.split("\\s\\|\\s");
+
+                String entryType = entries[0];
+                boolean isMark = entries[1].equals("1");
+                String entryText = entries[2];
+                String command;
+                if (entryType.equals("T")) {
+                    command = "todo " + entryText;
+                } else if (entryType.equals("D")) {
+                    command = "deadline " + entryText;
+                } else {
+                    command = "event " + entryText;
+                }
+
+                reAddTask(command);
+                if (isMark) {
+                    tasks.get(tasks.size() -1).setStatusIcon("mark");
+                }
+            }
+        }
+
+    }
+
+    public static void main(String[] args) throws FileNotFoundException {
+        boolean isGoodbye;
+        String command;
+
+        showWelcomeMessage();
+        load();
+
+        Scanner in = new Scanner(System.in);
+
+        // Loop that stops only when user enter "bye"
+        do {
+            command = in.nextLine();
+            isGoodbye = command.contains("bye");
+
+            // If user select list, show all the task
```

Comment:
See if this comment is necessary, as the code block performs many other functions and list is just one of them.

Code Quality: Comment minimally but sufficiently